dozer/cogs/info.py

```diff
@@ -10,48 +12,85 @@
-            activity_name = member.activity.type.name.replace("listening", "listening to") if member.activity else None
-
-            e = discord.Embed(color=member.color)
-            e.set_thumbnail(url=icon_url)
-            e.add_field(name='Name', value=str(member))
-            e.add_field(name='ID', value=member.id)
-            e.add_field(name='Nickname', value=member.nick, inline=member.nick is None)
-            e.add_field(name='Bot Created' if member.bot else 'User Joined Discord',
-                        value=member.created_at.strftime(datetime_format))
-            e.add_field(name='Joined Guild', value=member.joined_at.strftime(datetime_format))
-            e.add_field(name='Color', value=str(member.color).upper())
-
-            e.add_field(name='Status and Game', value=f'{member.status}, '.title() + (
-                f'{activity_name}' + f' {member.activity.name}' if member.activity else 'no game playing'), inline=False)
-            roles = reversed(member.roles[1:])  # Remove @everyone
-            e.add_field(name='Roles', value=', '.join(role.name for role in roles) or "No roles", inline=False)
-            e.add_field(name='Icon URL', value=icon_url, inline=False)
-        await ctx.send(embed=e)
+    @bot_has_permissions(embed_links=True)
+    async def member(self, ctx, *, member: discord.Member = None):
+        """ Show information on a member of this guild.
+            If no member is specified, your information will be shown."""
+        if member is None:
+            member = ctx.author
+
+        icon_url = member.avatar_url_as(static_format='png')
+
+        embed = discord.Embed(title=member.display_name, description=f'{member!s} ({member.id})', color=member.color)
+        embed.add_field(name='Bot Created' if member.bot else 'Account Created',
+                        value=member.created_at.strftime(self.datetime_format), inline=True)
+        embed.add_field(name='Member Joined', value=member.joined_at.strftime(self.datetime_format), inline=True)
+        if member.premium_since is not None:
+            embed.add_field(name='Member Boosted', value=member.premium_since.strftime(self.datetime_format), inline=True)
+        embed.add_field(name='Color', value=str(member.color).upper(), inline=True)
+
+        status = 'DND' if member.status is discord.Status.dnd else member.status.name.title()
+        if member.status is not discord.Status.offline:
+            platforms = self.pluralize([platform for platform in ('web', 'desktop', 'mobile') if
+                                        getattr(member, f'{platform}_status') is not discord.Status.offline])
+            status = f'{status} on {platforms}'
+        activities = '\n'.join(self._format_activities(member.activities))
+        embed.add_field(name='Status and Activity', value=f'{status}\n{activities}', inline=True)
+
+        embed.add_field(name='Roles', value=', '.join(role.name for role in member.roles[:0:-1]) or 'None')
```

Comment:
This field overflowing is something we might want to prepare for in the future. For now, the one closest to overflowing is a test account with every role, and that only reaches 900, so this PR isn't blocked on handling >1024 characters correctly.

Reply:
This is code I got from you :stuck_out_tongue:

Reply:
I'm no less fallible than anyone else, and I critique my own code as hard if not harder than anyone else's. 😉 But since this is a non-issue, I'm marking it as resolved.